Reduce set of build/tests which run on PRs.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#20775 Reduce set of build/tests which run on PRs.**

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D15499918](https://our.internmc.facebook.com/intern/diff/D15499918)

ezyang committed:
Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang commented:
Running builds after waiting a few minutes:

```
ci/circleci: AAA_pytorch_linux_xenial_cuda9_cudnn7_py2_test Pending — CircleCI is running your tests
ci/circleci: binary_linux_conda_2.7_cpu_build Pending — CircleCI is running your tests
ci/circleci: binary_linux_manywheel_2.7mu_cpu_devtoolset3_build Pending — CircleCI is running your tests
ci/circleci: binary_linux_manywheel_3.7m_cu100_devtoolset3_build Pending — CircleCI is running your tests
ci/circleci: binary_macos_conda_2.7_cpu_build Pending — CircleCI is running your tests
ci/circleci: binary_macos_libtorch_2.7_cpu_build Pending — CircleCI is running your tests
ci/circleci: binary_macos_wheel_3.6_cpu_build Pending — CircleCI is running your tests
ci/circleci: caffe2_cmake_cuda9_0_cudnn7_ubuntu16_04_build Pending — CircleCI is running your tests
ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_build Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_android_ubuntu16_04_build Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_clang7_ubuntu16_04_build Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_test Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_test Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_build Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_ios_macos10_13_build Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_mkl_ubuntu16_04_build Pending — CircleCI is running your tests
ci/circleci: caffe2_py2_system_macos10_13_build Pending — CircleCI is running your tests
ci/circleci: pytorch_linux_trusty_py2_7_9_build Pending — CircleCI is running your tests
ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_build Pending — CircleCI is running your tests
ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_test Pending — CircleCI is running your tests
ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_build Pending — CircleCI is running your tests
ci/circleci: pytorch_linux_xenial_py3_clang5_asan_build Pending — CircleCI is running your tests
ci/circleci: pytorch_macos_10_13_cuda9_2_cudnn7_py3_build Pending — CircleCI is running your tests
ci/circleci: pytorch_macos_10_13_py3_build Pending — CircleCI is running your tests
ci/circleci: pytorch_xla_linux_trusty_py3_6_gcc5_4_build Pending — CircleCI is running your tests
continuous-integration/travis-ci/pr Pending — The Travis CI build is in progress
gpr/caffe2-py2-clang7-rocmdeb-ubuntu16.04-test Pending — Build in progress
gpr/caffe2-py2-cuda9.0-cudnn7-windows-build Pending — Build in progress
gpr/caffe2-py2-devtoolset7-rocmrpm-centos7.5-test Pending — Build in progress
gpr/py2-clang7-rocmdeb-ubuntu16.04 Pending — Build in progress
gpr/py2-devtoolset7-rocmrpm-centos7.5 Pending — Build in progress
gpr/pytorch-win-ws2016-cuda9-cudnn7-py3 Pending — Build in progress
```

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

pjh5 commented on .circleci/verbatim-sources/linux-binary-build-defaults.yml:
```diff
@@ -58,6 +60,8 @@ binary_linux_test: &binary_linux_test
   - attach_workspace:
       at: /home/circleci/project
+  - run:
+      <<: *should_run_job
```
I don't think you need this or the one on upload. upload jobs always follow test jobs always follow build jobs

ezyang replied:
Actually I do, because `circleci step halt` only halts the step in the workflow; CircleCI will attempt to run the next step (test, upload). So need to cancel in all cases :)

pjh5 replied:
:/

pjh5 commented on .circleci/verbatim-sources/nightly-build-smoke-tests-defaults.yml:
```diff
@@ -12,6 +12,8 @@ smoke_linux_test: &smoke_linux_test
   - attach_workspace:
       at: /home/circleci/project
+  - run:
+      <<: *should_run_job
```
These will never run on PRs or on master, so these aren't needed either. These are the jobs that run everyday, download pytorch with the official pip/conda instructions, and then run smoke tests on them. They are thus only affected by the binaries of the previous night and never need to be run on master or PRs

pjh5 commented on .circleci/scripts/should_run_job.py:
```diff
@@ -0,0 +1,84 @@
+
+commit_msg = subprocess.check_output(("git", "log", "--format=%B", "-n", "1", "HEAD"), universal_newlines=True)
+
+RE_MARKER = re.compile(r'\[(?:([^ \[\]]+) )?(?:ci|test)(?: ([^ \[\]]+))?\]')
```
Can you leave a comment for posterity?

pjh5 commented on .circleci/scripts/should_run_job.py:
```diff
@@ -0,0 +1,84 @@
+        continue
+    spec = m.group(1) or m.group(2)
+    if spec in args.build_environment or spec == 'all':
+        print("Accepting {} due to commit marker {}".format(args.build_environment, m.group(0)))
```
IIUC this looks for a string in the commit message and runs all the jobs if it finds the right thing. I don't think this will work on the nightly jobs that always take whatever the last commit at midnight EST is. There is a way to detect if you are a scheduled build (triggered via timer) vs master/PR ; one of the CircleCI git related env variables will be missing on the scheduled builds. It's impossible to test this without merging the change though.

ezyang replied:
The test is in a shell script which I forgot to check in. Incoming!

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

pjh5 commented on .circleci/scripts/should_run_job.sh:
```diff
@@ -0,0 +1,16 @@
+# Check if we should actually run
+echo "BUILD_ENVIRONMENT: ${BUILD_ENVIRONMENT}"
+echo "CIRCLE_PULL_REQUEST: ${CIRCLE_PULL_REQUEST:-}"
+if ! [ -z "${CIRCLE_PULL_REQUEST:-}" ]; then
```
Do you have a reason for ``` if ! [ -z "..." ]; then``` vs ```if [[ -n "..." ]]; then``` ?

ezyang replied:
The proximal reason is that the old code did it this way, and I just copy pasted it. The original reason is probably, I could remember what `-z` was, I couldn't remember what the inverse was, so I just negated it.

pjh5 commented on .circleci/scripts/should_run_job.sh:
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -ex -o pipefail
```
Can you add -u for no reads from unassigned bash variables? You'll need to change -z "$CIRCLE_PULL_REQUEST" to -z "${CIRCLE_PULL_REQUEST:-}"

pjh5 commented on .circleci/config.yml:
```diff
@@ -130,10 +130,23 @@ install_doc_push_script: &install_doc_push_script
+    elif [ -e /Users/distiller/pytorch-ci-env/workspace/.circleci/scripts/should_run_job.sh ]; then
+      /Users/distiller/pytorch-ci-env/workspace/.circleci/scripts/should_run_job.sh
+    else
+      .circleci/scripts/should_run_job.sh
```
Can you also add a comment in each clause for which executor it's for

pjh5 replied:
Although I thought that ~/workspace would work on both macos and machine executors

ezyang commented:
Most recent build had the following runs and skips:

```
MacBook-Pro-116:ghstack ezyang$ ghstack status https://github.com/pytorch/pytorch/pull/20775
RAN ci/circleci: caffe2_cmake_cuda9_0_cudnn7_ubuntu16_04_build
RAN ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_build
RAN ci/circleci: caffe2_py2_android_ubuntu16_04_build
RAN ci/circleci: caffe2_py2_clang7_ubuntu16_04_build
RAN ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_build
RAN ci/circleci: caffe2_py2_mkl_ubuntu16_04_build
RAN ci/circleci: caffe2_py2_system_macos10_13_build
RAN ci/circleci: pytorch_linux_trusty_py2_7_9_build
RAN ci/circleci: pytorch_linux_trusty_py2_7_9_test
RAN ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_build
RAN ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_test
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX2_test
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX_NO_AVX2_test
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_build
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_multigpu_test
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_nogpu_test
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_slow_test
RAN ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_test
RAN ci/circleci: pytorch_linux_xenial_py3_clang5_asan_build
RAN ci/circleci: pytorch_linux_xenial_py3_clang5_asan_test
RAN ci/circleci: pytorch_macos_10_13_py3_build
RAN ci/circleci: pytorch_macos_10_13_py3_test
RAN ci/circleci: pytorch_xla_linux_trusty_py3_6_gcc5_4_build
RAN ci/circleci: pytorch_xla_linux_trusty_py3_6_gcc5_4_test
RAN ci/circleci: setup
RAN continuous-integration/travis-ci/pr
RAN pr/caffe2-py2-clang7-rocmdeb-ubuntu16.04-test
RAN pr/caffe2-py2-cuda9.0-cudnn7-windows-build
RAN pr/caffe2-py2-devtoolset7-rocmrpm-centos7.5-test
RAN pr/py2-clang7-rocmdeb-ubuntu16.04
RAN pr/py2-devtoolset7-rocmrpm-centos7.5
RAN pr/pytorch-win-ws2016-cuda9-cudnn7-py3
SKIPPED ci/circleci: AAA_pytorch_linux_trusty_py3_5_build
SKIPPED ci/circleci: AAA_pytorch_linux_trusty_py3_5_test
SKIPPED ci/circleci: AAA_pytorch_linux_xenial_cuda9_cudnn7_py2_build
SKIPPED ci/circleci: AAA_pytorch_linux_xenial_cuda9_cudnn7_py2_test
SKIPPED ci/circleci: binary_linux_conda_2.7_cpu_build
SKIPPED ci/circleci: binary_linux_conda_2.7_cpu_test
SKIPPED ci/circleci: binary_linux_manywheel_2.7mu_cpu_devtoolset3_build
SKIPPED ci/circleci: binary_linux_manywheel_2.7mu_cpu_devtoolset3_test
SKIPPED ci/circleci: binary_linux_manywheel_3.7m_cu100_devtoolset3_build
SKIPPED ci/circleci: binary_linux_manywheel_3.7m_cu100_devtoolset3_test
SKIPPED ci/circleci: binary_macos_conda_2.7_cpu_build
SKIPPED ci/circleci: binary_macos_libtorch_2.7_cpu_build
SKIPPED ci/circleci: binary_macos_wheel_3.6_cpu_build
SKIPPED ci/circleci: caffe2_cmake_cuda9_0_cudnn7_ubuntu16_04_test
SKIPPED ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_test
SKIPPED ci/circleci: caffe2_py2_clang3_8_ubuntu16_04_build
SKIPPED ci/circleci: caffe2_py2_clang3_9_ubuntu16_04_build
SKIPPED ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_build
SKIPPED ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_test
SKIPPED ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_build
SKIPPED ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_test
SKIPPED ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_test
SKIPPED ci/circleci: caffe2_py2_gcc4_8_ubuntu14_04_build
SKIPPED ci/circleci: caffe2_py2_gcc4_8_ubuntu14_04_test
SKIPPED ci/circleci: caffe2_py2_ios_macos10_13_build
SKIPPED ci/circleci: caffe2_py2_mkl_ubuntu16_04_test
SKIPPED ci/circleci: pytorch_doc_push
SKIPPED ci/circleci: pytorch_linux_trusty_py2_7_build
SKIPPED ci/circleci: pytorch_linux_trusty_py2_7_test
SKIPPED ci/circleci: pytorch_linux_trusty_py3_6_gcc4_8_build
SKIPPED ci/circleci: pytorch_linux_trusty_py3_6_gcc4_8_test
SKIPPED ci/circleci: pytorch_linux_trusty_py3_6_gcc7_build
SKIPPED ci/circleci: pytorch_linux_trusty_py3_6_gcc7_test
SKIPPED ci/circleci: pytorch_linux_trusty_pynightly_build
SKIPPED ci/circleci: pytorch_linux_trusty_pynightly_test
SKIPPED ci/circleci: pytorch_linux_xenial_cuda10_cudnn7_py3_gcc7_build
SKIPPED ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_build
SKIPPED ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_test
SKIPPED ci/circleci: pytorch_linux_xenial_py3_clang5_android_ndk_r19c_build
SKIPPED ci/circleci: pytorch_macos_10_13_cuda9_2_cudnn7_py3_build
SKIPPED ci/circleci: pytorch_namedtensor_linux_trusty_py3_6_gcc5_4_build
SKIPPED ci/circleci: pytorch_namedtensor_linux_trusty_py3_6_gcc5_4_test
SKIPPED ci/circleci: pytorch_short_perf_test_gpu
```

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Fix working directory bug on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

pjh5 commented on .circleci/verbatim-sources/linux-binary-build-defaults.yml:
```diff
@@ -54,10 +57,11 @@ binary_linux_test: &binary_linux_test
   - attach_workspace:
       at: ~/workspace
-  - attach_workspace:
-      at: /home/circleci/project
```
This is what's breaking https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/binary_run_in_docker.sh#L33

pjh5 replied:
it should use https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/binary_populate_env.sh#L87 , but just harcode it to the right thing for now.

We should really unify the ~/workspace and binary $workdir. I think it's not this way now cause ~/workspace == pytorch iiuc, but in binaries there's two repos so we need ~/workspace/pytorch and ~/workspace/builder

ezyang committed:
Update on "Reduce set of build/tests which run on PRs."

Reduce set of build/tests which run on PRs.

I ran an analysis of historical build data to find out which
build jobs success/failures where correlated with each other.
The analysis can be found here:

  https://github.com/ezyang/pytorch-ci-deduplication/blob/master/PyTorch%20CI%20deduplication%20(1).pdf

Based on this analysis, I've revamped our logic for whether or not to
run a PR or not to only run builds in our default set.  This replaces
our previous logic for doing [ci slow] or [ci xla]; now, you can place
any string [ci foobar] and it will cause any build which matches that
substring to be run on your run.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 20775 gh/ezyang/134/head

ezyang commented:
Final status, it looks good to me:

```
MacBook-Pro-116:ghstack ezyang$ ghstack status https://github.com/pytorch/pytorch/pull/20775
✅ ci/circleci: binary_linux_manywheel_2.7mu_cpu_devtoolset3_build        https://circleci.com/gh/pytorch/pytorch/1793432
✅ ci/circleci: binary_linux_manywheel_2.7mu_cpu_devtoolset3_test         https://circleci.com/gh/pytorch/pytorch/1793786
✅ ci/circleci: caffe2_cmake_cuda9_0_cudnn7_ubuntu16_04_build             https://circleci.com/gh/pytorch/pytorch/1793454
✅ ci/circleci: caffe2_cmake_cuda9_0_cudnn7_ubuntu16_04_test              https://circleci.com/gh/pytorch/pytorch/1794246
✅ ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_build                    https://circleci.com/gh/pytorch/pytorch/1793464
✅ ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_test                     https://circleci.com/gh/pytorch/pytorch/1793766
✅ ci/circleci: caffe2_py2_android_ubuntu16_04_build                      https://circleci.com/gh/pytorch/pytorch/1793444
✅ ci/circleci: caffe2_py2_clang7_ubuntu16_04_build                       https://circleci.com/gh/pytorch/pytorch/1793463
✅ ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_build               https://circleci.com/gh/pytorch/pytorch/1793459
✅ ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_test                https://circleci.com/gh/pytorch/pytorch/1794006
✅ ci/circleci: caffe2_py2_mkl_ubuntu16_04_build                          https://circleci.com/gh/pytorch/pytorch/1793434
✅ ci/circleci: caffe2_py2_mkl_ubuntu16_04_test                           https://circleci.com/gh/pytorch/pytorch/1793773
✅ ci/circleci: caffe2_py2_system_macos10_13_build                        https://circleci.com/gh/pytorch/pytorch/1793447
✅ ci/circleci: pytorch_doc_push                                          https://circleci.com/gh/pytorch/pytorch/1794700
✅ ci/circleci: pytorch_linux_trusty_py2_7_9_build                        https://circleci.com/gh/pytorch/pytorch/1793441
✅ ci/circleci: pytorch_linux_trusty_py2_7_9_test                         https://circleci.com/gh/pytorch/pytorch/1793775
✅ ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_build                   https://circleci.com/gh/pytorch/pytorch/1793449
✅ ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_test                    https://circleci.com/gh/pytorch/pytorch/1794187
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX2_test        https://circleci.com/gh/pytorch/pytorch/1794699
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_NO_AVX_NO_AVX2_test https://circleci.com/gh/pytorch/pytorch/1794696
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_build               https://circleci.com/gh/pytorch/pytorch/1793465
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_multigpu_test       https://circleci.com/gh/pytorch/pytorch/1794693
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_nogpu_test          https://circleci.com/gh/pytorch/pytorch/1794697
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_slow_test           https://circleci.com/gh/pytorch/pytorch/1794698
✅ ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_test                https://circleci.com/gh/pytorch/pytorch/1794695
✅ ci/circleci: pytorch_linux_xenial_py3_clang5_android_ndk_r19c_build    https://circleci.com/gh/pytorch/pytorch/1793457
✅ ci/circleci: pytorch_linux_xenial_py3_clang5_asan_build                https://circleci.com/gh/pytorch/pytorch/1793450
✅ ci/circleci: pytorch_linux_xenial_py3_clang5_asan_test                 https://circleci.com/gh/pytorch/pytorch/1793918
✅ ci/circleci: pytorch_short_perf_test_gpu                               https://circleci.com/gh/pytorch/pytorch/1794694
✅ ci/circleci: setup                                                     https://circleci.com/gh/pytorch/pytorch/1793429
✅ continuous-integration/travis-ci/pr                                    https://travis-ci.org/pytorch/pytorch/builds/536899117?utm_source=github_status&utm_medium=notification
❌ ci/circleci: pytorch_macos_10_13_cuda9_2_cudnn7_py3_build              https://circleci.com/gh/pytorch/pytorch/1793462
❌ ci/circleci: pytorch_xla_linux_trusty_py3_6_gcc5_4_build               https://circleci.com/gh/pytorch/pytorch/1793433
❌ pr/caffe2-py2-cuda9.0-cudnn7-windows-build                             https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda9.0-cudnn7-windows-trigger-build/49843/
❔ ci/circleci: AAA_pytorch_linux_trusty_py3_5_build                      https://circleci.com/gh/pytorch/pytorch/1793439
❔ ci/circleci: AAA_pytorch_linux_trusty_py3_5_test                       https://circleci.com/gh/pytorch/pytorch/1793485
❔ ci/circleci: AAA_pytorch_linux_xenial_cuda9_cudnn7_py2_build           https://circleci.com/gh/pytorch/pytorch/1793451
❔ ci/circleci: AAA_pytorch_linux_xenial_cuda9_cudnn7_py2_test            https://circleci.com/gh/pytorch/pytorch/1793487
❔ ci/circleci: binary_linux_conda_2.7_cpu_build                          https://circleci.com/gh/pytorch/pytorch/1793460
❔ ci/circleci: binary_linux_conda_2.7_cpu_test                           https://circleci.com/gh/pytorch/pytorch/1793493
❔ ci/circleci: binary_linux_manywheel_3.7m_cu100_devtoolset3_build       https://circleci.com/gh/pytorch/pytorch/1793461
❔ ci/circleci: binary_linux_manywheel_3.7m_cu100_devtoolset3_test        https://circleci.com/gh/pytorch/pytorch/1793492
❔ ci/circleci: binary_macos_conda_2.7_cpu_build                          https://circleci.com/gh/pytorch/pytorch/1793452
❔ ci/circleci: binary_macos_libtorch_2.7_cpu_build                       https://circleci.com/gh/pytorch/pytorch/1793435
❔ ci/circleci: binary_macos_wheel_3.6_cpu_build                          https://circleci.com/gh/pytorch/pytorch/1793440
❔ ci/circleci: caffe2_py2_clang3_8_ubuntu16_04_build                     https://circleci.com/gh/pytorch/pytorch/1793448
❔ ci/circleci: caffe2_py2_clang3_9_ubuntu16_04_build                     https://circleci.com/gh/pytorch/pytorch/1793436
❔ ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_build                   https://circleci.com/gh/pytorch/pytorch/1793453
❔ ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_test                    https://circleci.com/gh/pytorch/pytorch/1793478
❔ ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_build               https://circleci.com/gh/pytorch/pytorch/1793458
❔ ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_test                https://circleci.com/gh/pytorch/pytorch/1793486
❔ ci/circleci: caffe2_py2_gcc4_8_ubuntu14_04_build                       https://circleci.com/gh/pytorch/pytorch/1793443
❔ ci/circleci: caffe2_py2_gcc4_8_ubuntu14_04_test                        https://circleci.com/gh/pytorch/pytorch/1793481
❔ ci/circleci: caffe2_py2_ios_macos10_13_build                           https://circleci.com/gh/pytorch/pytorch/1793466
❔ ci/circleci: pytorch_linux_trusty_py2_7_build                          https://circleci.com/gh/pytorch/pytorch/1793431
❔ ci/circleci: pytorch_linux_trusty_py2_7_test                           https://circleci.com/gh/pytorch/pytorch/1793484
❔ ci/circleci: pytorch_linux_trusty_py3_6_gcc4_8_build                   https://circleci.com/gh/pytorch/pytorch/1793442
❔ ci/circleci: pytorch_linux_trusty_py3_6_gcc4_8_test                    https://circleci.com/gh/pytorch/pytorch/1793483
❔ ci/circleci: pytorch_linux_trusty_py3_6_gcc7_build                     https://circleci.com/gh/pytorch/pytorch/1793438
❔ ci/circleci: pytorch_linux_trusty_py3_6_gcc7_test                      https://circleci.com/gh/pytorch/pytorch/1793482
❔ ci/circleci: pytorch_linux_trusty_pynightly_build                      https://circleci.com/gh/pytorch/pytorch/1793445
❔ ci/circleci: pytorch_linux_trusty_pynightly_test                       https://circleci.com/gh/pytorch/pytorch/1793480
❔ ci/circleci: pytorch_linux_xenial_cuda10_cudnn7_py3_gcc7_build         https://circleci.com/gh/pytorch/pytorch/1793437
❔ ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_build        https://circleci.com/gh/pytorch/pytorch/1793456
❔ ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_test         https://circleci.com/gh/pytorch/pytorch/1793479
❔ ci/circleci: pytorch_macos_10_13_py3_build                             https://circleci.com/gh/pytorch/pytorch/1793446
❔ ci/circleci: pytorch_macos_10_13_py3_test                              https://circleci.com/gh/pytorch/pytorch/1793468
❔ ci/circleci: pytorch_namedtensor_linux_trusty_py3_6_gcc5_4_build       https://circleci.com/gh/pytorch/pytorch/1793455
❔ ci/circleci: pytorch_namedtensor_linux_trusty_py3_6_gcc5_4_test        https://circleci.com/gh/pytorch/pytorch/1793477
🚸 pr/caffe2-py2-clang7-rocmdeb-ubuntu16.04-test                          https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/31453/
🚸 pr/caffe2-py2-devtoolset7-rocmrpm-centos7.5-test                       https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger-test/24462/
🚸 pr/py2-clang7-rocmdeb-ubuntu16.04                                      https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger/31421/
🚸 pr/py2-devtoolset7-rocmrpm-centos7.5                                   https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger/16712/
🚸 pr/pytorch-win-ws2016-cuda9-cudnn7-py3                                 https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-win-ws2016-cuda9-cudnn7-py3-trigger/54427/
```
